src/app/problems/_test/mock.ts

```diff
@@ -5,6 +5,76 @@ import {McqFormData} from "@app/problems/_forms/mcq.form"
 import {JavaFormData} from "@app/problems/_forms/java.form"
 import {ParsonsFormData} from "@app/problems/_forms/parsons.form"
 import {UserStats} from "@app/_models/user_difficulty_stats"
+import {Goal, GoalItem} from "@app/_models/goal/goal"
+import {GoalForm, GoalFormData, GoalItemFormData} from "@app/course/_forms/goal.form"
+import {FormArray, FormBuilder, FormControl, FormGroup, Validators} from "@angular/forms"
+import {TuiDay, TuiTime} from "@taiga-ui/cdk"
+
+const builder = new FormBuilder()
+
+export const MOCK_FORM: FormGroup = builder.group({
+    end_date: new FormControl(TuiDay.currentLocal(), [Validators.required]),
+    end_time: new FormControl(TuiTime.currentLocal(), [Validators.required]),
+    goal_items: new FormArray([GoalForm.createGoalItemForm()], [Validators.required]),
+})
+
+export const MOCK_GOAL_FORM: GoalFormData = {
+    course_id: 1,
+    end_date: "2022-11-10T16:40:37.860661-08:00"
+}
+
+export const MOCK_GOAL_ITEM_FORM: GoalItemFormData = {
+    goal: 3,
```

Comment:
The mock data should makes sense. It should not be completely random.
Your Goal mocks have 13 and 19 as their ids. This goal item is connected to a goal with id 3 which does not exist in the mocks.

Same for category...